rust/parquet/src/arrow/arrow_writer.rs

```diff
@@ -855,13 +937,14 @@ mod tests {
     }
 
     #[test]
+    #[ignore = "list support is incomplete"]
     fn arrow_writer_complex() {
         // define schema
         let struct_field_d = Field::new("d", DataType::Float64, true);
         let struct_field_f = Field::new("f", DataType::Float32, true);
         let struct_field_g = Field::new(
             "g",
-            DataType::List(Box::new(Field::new("items", DataType::Int16, false))),
```

Comment:
FYIW the PR from @ch-sc in https://github.com/apache/arrow/pull/8715 will almost certainly conflict with this PR

Reply:
Tests still pass after rebasing, so that's great. CC @jorgecarleitao